### PR TITLE
LoupeTarget with support for include MessageDetails in Json-format

### PR DESCRIPTION
**Pull request by snakefoot**
Added `IncludeEventProperties` + `IncludeMldc` + `ContextProperties`.

Possible to do this:

```xml
<target name="Loupe" xsi:type="Loupe" includeEventProperties="true">
    <contextProperty name="threadid" layout="${threadid}" />
</target>
```

This will match `Loupe.Serilog` way of resolving details by including LogEvent-Properties.

### CODE REVIEW

File: src/Agent.NLog45/Agent.NLog45.csproj [hunk 1]
```diff
@@ -37,7 +37,7 @@
       <Version>4.9.0.13</Version>
     </PackageReference>
     <PackageReference Include="NLog">
-      <Version>4.5.0</Version>
+      <Version>4.5.11</Version>
```
Comment (kendallmiller) [inline src/Agent.NLog45/Agent.NLog45.csproj:40]: Is 4.5.11 the minimum version of NLog necessary for this to work?  We generally try to have Loupe target the lowest version for highest compatibility.

**Comment by kendallmiller** [comment]
I definitely like getting this enhancement done, and double-checking it with how we did Loupe.Serilog.  Now I'm regretting we didn't go further and rename the NuGet package to Loupe.NLog and drop the "Agent".

File: src/Agent.NLog45/Agent.NLog45.csproj (hunk 1 above)
Comment (snakefoot) [reply src/Agent.NLog45/Agent.NLog45.csproj:40]: NLog 4.5.11 is from year 2018, and includes bug-fixes for the JsonSerializer: https://www.nuget.org/packages/NLog/4.5.11

**Comment by snakefoot** [comment]
Did a force-push because I forgot to include `OptimizeBufferReuse = true`.

**Comment by snakefoot** [comment]
@kendallmiller Should I rebase to `develop`-branch ?

**Comment by snakefoot** [comment]
> Now I'm regretting we didn't go further and rename the NuGet package to Loupe.NLog and drop the "Agent".

It seems like you also have [Loupe.Agent.AspNetCore](https://www.nuget.org/packages/Loupe.Agent.AspNetCore/), but then again you also have [Loupe.Extensions.Logging](https://www.nuget.org/packages/Loupe.Extensions.Logging/). Two naming conventions is better than one?

**Comment by kendallmiller** [comment]
> @kendallmiller Should I rebase to `develop`-branch ?

That's probably a good idea.

**Comment by kendallmiller** [comment]
We have Loupe.Serilog.Core and Loupe.Serilog packages to differentiate the .NET Standard/Core from the .NET Framework versions since the Loupe Agent dependency is different.  

Overall, the number of packages on the .NET Core side exploded more than I'd have liked:  To keep Microsoft.Extensions.Logging as an optional dependency introduced a few.

If you look at the Develop branch you can see we've started a grand refactor of the Agent to enable more scenarios and somewhat reduce the package count, I think we'll make that the .NET 5 version.

**Comment by snakefoot** [comment]
> > Should I rebase to develop-branch ?

> That's probably a good idea.

Can you merge the changes made to master back to develop? (Ex.  "Modified target naming")

**Comment by kendallmiller** [comment]
On it

**Comment by kendallmiller** [comment]
Done

**Comment by snakefoot** [comment]
Replaced by #13

**Pull request closed without merging**